jOOQ/src/main/java/org/jooq/impl/DefaultBinding.java

```diff
@@ -2389,6 +2391,29 @@ private static final String preparse(String formatted, boolean includeDate) {
 
         // Probably a bug, let OffsetDateTime or OffsetTime report it
         else {
+
+            // [#7126] Postgresql can return Special Date/Time Inputs such as "epoch", "infinity", "now", ...
+            if ("epoch".equals(string)) {
+                return "1970-01-01T00:00Z";
+            } else if ("infinity".equals(string)) {
+                // DATE_POSITIVE_INFINITY taken from org.postgresql.PGStatement
+                return "+292278994-08-16T23:00Z";
+            } else if ("-infinity".equals(string)) {
+                // DATE_NEGATIVE_INFINITY taken from org.postgresql.PGStatement
+                return "-292275055-05-16T23:00Z";
+            } else if ("now".equals(string)) {
+                return OffsetDateTime.now().toString();
+            } else if ("today".equals(string)) {
+                return OffsetDateTime.ofInstant(LocalDate.now().atStartOfDay(ZoneId.systemDefault()).toInstant(), ZoneId.systemDefault()).toString();
```

Comment:
... likewise, calling `now()` or `ZoneId.systemDefault()` seams treacherous, as we're going to get the *client* default, not the *server* default. I had rather these continue failing than getting this wrong.

I agree with the epoch / infinity ones, though

Reply:
Hallo Lukas, 
Please excuse me for this, I wrote the patch without testing it deeply. Actually the only two values which are returned as text (and not as ISO 8601) are `"-infinity"` and `"infinity"`. All the other values are returned in ISO 8601 format, using the time zone of the Postgresql server.

Therefore you can delete all the problematic lines (i.e. `"epoch"`, `"now"`, `"today"`, `"tomorrow"`, `"yesterday"`, `"allballs"`) since the server will make all the computations.

    create table timezoneissue
    (
        v timestamp with time zone
    );
    
    insert into timezoneissue(v) values ('-infinity'::timestamptz), ('infinity'::timestamptz), ('epoch'::timestamptz), ('now'::timestamptz), ('today'::timestamptz), ('tomorrow'::timestamptz), ('yesterday'::timestamptz); --('allballs'::timestamptz);

    select x.v from timezoneissue x;

Reply:
I see, thanks for the clarification

Reply:
Also `"epoch"`can be deleted, since the JDBC driver will return a nice `"1970-01-01 01:00:00+01"` string 😃